notebooks/poisonous_mushroom_classifier.qmd

```
@@ -447,17 +445,87 @@ Before we continue, we must preprocess our data such that the features work with
 
 preprocessor = OneHotEncoder()
```

Great job! Please add `...handle_unknown="ignore"` to `OneHotEncoder()` as in the code below: `OneHotEncoder(handle_unknown="ignore")`

---

Thanks, Ruth. I’m wondering whether we actually want the model to handle unknown categories silently. It might be safer for the model to raise an error whenever it encounters an unexpected feature value, since we don’t anticipate any unknowns in the input, and if a new prediction has unknown values, that should be corrected I think. My instinct is that for a poisonous–mushroom classifier, silently ignoring unknown categories could be risky, because it may lead to incorrect predictions in a health-critical context. Ensuring the model only makes predictions on expected, validated feature values seems like a safer approach